inc/endpoints/class-components-endpoint.php

```diff
@@ -358,6 +361,34 @@ public function add_api_link_to_terms( array $actions, \WP_Term $term ) : array
 		);
 		return $actions;
 	}
+
+	/**
+	 * Force trailing slashes on all non-404, and non-file requests.
+	 */
+	public function force_trailing_slashes() {
+
+		// Return if there is already a trailing slash, or no posts were
+		// returned by the path's query.
+		if (
+			trailingslashit( $this->path ) === $this->path
+			|| ! $this->query->have_posts()
+		) {
+			return;
+		}
+
+		// Apply a trailing slash to the path.
+		$this->params['path'] = trailingslashit( $this->path );
+
+		// Apply all params to the rest url for this endpoint.
+		$request_url = add_query_arg(
+			$this->params,
+			rest_url( 'irving/v1/components' )
+		);
+
+		// Redirect permanently.
+		wp_redirect( $request_url, 301 );
+		exit();
```

Comment:
Technically exit is a language construct, not a function, so you don't need parens.